plugin/session_buffer.py

```diff
@@ -143,6 +160,12 @@ def language_id(self) -> str:
     def add_session_view(self, sv: SessionViewProtocol) -> None:
         self.session_views.add(sv)
 
+    def remove_session_view(self, sv: SessionViewProtocol) -> None:
+        for scope in self.semantic_tokens.active_scopes.keys():
+            key = "lsp_{}".format(scope)
+            sv.view.erase_regions(key)
+        self.session_views.remove(sv)
```

Comment:
Hmm, do we have to remove regions for each View (SessionView) or only for the last one closing? I think only the last one.

This seems to work but I can see that `remove_session_view` is not called in most cases (when closing views). It's only called when restarting session. So that might be the reason why it's working.

I would anyway suggest this instead:

```suggestion
        self.session_views.remove(sv)
        if not self.session_views:
            self._clear_semantic_regions()
```

and define `_clear_semantic_regions`.

Reply:
But if we only clear regions when the last SessionView is removed, then we don't have the references to the other SessionViews no more, so it would be the same problem as before? Not sure if I understand right, but what would you put in `_clear_semantic_regions()`?

Isn't the restart/shutdown session exactly the point we are interested in?

Reply:
Then 

```py
        self.session_views.remove(sv)
        if not self.session_views:
            self._clear_semantic_regions(sv.view)
```

In `_clear_semantic_regions` put the current code. I'd just prefer it separate from the `remove_session_view`.

> Isn't the restart/shutdown session exactly the point we are interested in?

Yes, that's all we actually need but ideally the `remove_session_view` would be called in all cases as expected.

Reply:
I can move it to a separate method, but I still don't understand the `not self.session_views` check. In this example you only provide the very last SessionView as argument, so how can we access and clear regions for other views of the buffer?

Reply:
That's the thing I was not sure about but it looks like regions separate for cloned views unlike Phantoms. So then we would need to reset them clear them for every invocation of `remove_session_view`, yes.